frappe/integrations/doctype/nextcloud_settings/nextcloud_settings.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe.model.document import Document
+from frappe import _
+from frappe.utils.backups import new_backup
+from frappe.utils.background_jobs import enqueue
+from frappe.integrations.offsite_backup_utils import get_latest_backup_file, send_email, validate_file_size
+from frappe.utils import cint, get_backups_path
+
+import requests
+import os
+from rq.timeouts import JobTimeoutException
+from urllib.parse import urlparse
+
+class NextCloudSettings(Document):
+	def validate(self):
+		if not self.enabled:
+			return
+
+		if self.enabled:
+			if (not self.email or self.email == "" 
+				or not self.password or self.password == "" 
+				or not self.domain_url or self.domain_url == "" 
+				or not self.webdav_url or self.webdav_url == ""
+				or not self.send_notifications_to or self.send_notifications_to == "" 
+				or not self.backup_frequency or self.backup_frequency == ""):
+				frappe.throw(_('If Nexcloud is enabled, the data is mandatory: email, Password, Domain URL, Webdav URL, Send Notifications To and Backup Frequency'))
```

Comment:
Check individual params separately, that way you can tell the user which values are missing.